authorizations/authorization-ranger/src/main/java/org/apache/gravitino/authorization/ranger/RangerHelper.java

```diff
@@ -480,8 +505,8 @@ protected void updatePolicyOwnerRole(RangerPolicy policy, String ownerRoleName)
               policyItem
                   .getAccesses()
                   .add(new RangerPolicy.RangerPolicyItemAccess(ownerPrivilege.getName()));
-              if (!policyItem.getRoles().contains(ownerRoleName)) {
-                policyItem.getRoles().add(ownerRoleName);
+              if (!policyItem.getRoles().contains(rangerRoleName(ownerRoleName))) {
+                policyItem.getRoles().add(rangerRoleName(ownerRoleName));
```

Comment:
Not sure if this is a good pattern.
When you do a `getRoles()` call, I'd guess that you are creating a copy of the roles.
The `.add()` call is then performed on your new copy.

To make the semantics clear and solid, I'd suggest we refactor policyItem to add a
`hasRole()` method and `addRole` method.

Reply:
`policyItem`  is from `ranger-plugins-common`. Its type is `org.apache.ranger.plugin.model.RangerPolicy.RangerPolicyItem.` 

RangerHelper is a tool that works directly with Ranger SDK, so I believe it's better to keep it simple and not refactor policyItem.

Reply:
Your current code may work as expected, what I was trying to point out is that the pattern may be improved. A better logic should be this:

```java
RangerRoles roles = policyItem.getRoles();
if (!roles.contains(ownerRoleName)) {
    roles.add(rangerRoleName(ownerRoleName));
    policyItem.setRoles(roles);
}
```

In other words, you do a GET and then you do a SET. The thing(s) you get from
`getRoles()` is a reference in Java. That is why your current code works.
So ... my point was not that your code is wrong. It is not a pattern for robust code IMHO.

Reply:
Thank you for your advice. I get it now and agree that it's a better pattern. 
However, there are a lot of places that would need to be rewritten. 
I'd like to focus on the core feature in this PR and handle the refactoring in a separate one.